lib/provisioning/Provisioner.js

```diff
@@ -341,6 +359,123 @@ Provisioner.prototype.handlePm = function(server, fromUser, text) {
     log.warn(`Provisioner was not expecting PM from ${fromUser.nick} on ${server.domain}`);
 }
 
+
+// Get information that might be useful prior to calling requestLink
+//  with the same parameters
+Provisioner.prototype.queryLink = Promise.coroutine(function*(options) {
+    let ircDomain = options.remote_room_server;
+    let ircChannel = options.remote_room_channel;
+    let roomId = options.matrix_room_id;
+    let opNick = options.op_nick;
+    let key = options.key || undefined; // Optional key
+    let userId = options.user_id;
+
+    let queryInfo = {
+        // Map between options and info pertaining to them
+        // $parameter {
+        //      info: 'some information' ,
+        //      error: 'an error'
+        // }
```

Comment:
You say this, but never seem to use the `info` property, only `error`.

It seems that this function is mainly doing validation, rather than giving any kind of useful information back to the caller (such as a list of operator nicks of whom you can contact, which iirc was the main reason for wanting this API in the first place).